### DISCUSSION TITLE
Operators as operands to ternary conditional operator

### DISCUSSION THREAD

**Discussion by ActivistInvestor** [dotnet/csharplang#7946] [Language Ideas]
A constant source of code expansion involves code that adds or removes handlers of events based on a condition.

For example:

```
bool enabled ^= true;
var events = PropertyInspector.EventManager;
if(enabled)
   events.propertyChanged += PropertyInspector_propertyChanged;
else
   events.propertyChanged -= PropertyInspector_propertyChanged;

```

Although this is probably out of the realm of possibility, it would be helpful if the ternary conditional operator could accept other operators as operands, so I could instead write:

```
bool enabled ^= true;
var events = PropertyInspector.EventManager;
events.propertyChanged (enabled ? += : -=) PropertyInspector_propertyChanged;

```

**Comment by TahirAhmadov** [comment]
This is very unlikely to happen, but you can write a source generator which generates methods for these events and does what you want:
```cs
[EventMethods]
partial class EventManager
{
  public event EventHandler PropertyChanged;
}
```
Your source generator would look for the attribute `[EventMethods]` and generate the following:
```cs
partial class EventManager
{
  public void PropertyChangedAddRemove(EventHandler eventHandler, bool enabled)
  {
    if(enabled)
      this.PropertyChanged += eventHandler;
    else
      this.PropertyChanged -= eventHandler;
  }
}
```
Then elsewhere:
```cs
bool enabled = true;
var events = PropertyInspector.EventManager;
events.PropertyChangedAddRemove(PropertyInspector_propertyChanged, enabled);
```

**Reply by ActivistInvestor** [reply]
Thanks, I currently use an extension method that deals with this in pretty much the same way, since in some cases the class that I have to add and remove events to/from is not mine, and therefore have no control over it.
